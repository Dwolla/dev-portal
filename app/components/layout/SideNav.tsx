import React, { useEffect, useState } from "react";
import { Button } from "@mui/material";
// import { styled } from '@mui/material/styles';
import { useRouter } from "next/router";
import Link from "next/link";
import classnames from "classnames";
import groupby from "lodash.groupby";
import styled from "@emotion/styled";
import { css } from "@emotion/react";
import sortBy from "lodash.sortby";
import uniqBy from "lodash.uniqby";
import {
  TopBarProps as MobileItemProps, // eslint-disable-line no-unused-vars
} from "./TopBar";
// import Button from "../base/Button";
import {
  ORANGE_PRIMARY,
  WHITE_PRIMARY,
  GREY_2,
  PURPLE_PRIMARY,
  LAYOUT_BORDER,
  GREY_HOVER,
  VERTICAL_GRADIENT,
} from "../colors";
import { POPPINS, ROBOTO } from "../typography";
import { breakUp, breakDown } from "../breakpoints";
import { slideInFromLeft, slideInFromRight } from "../keyframes";
import { ReactComponent as BackIcon } from "../../../assets/images/component-icons/side-nav/back-nav-icon.svg";
import { ReactComponent as RightIcon } from "../../../assets/images/component-icons/side-nav/caret-right-nav-icon.svg";
import { ReactComponent as NewTabIcon } from "../../../assets/images/component-icons/open-in-new-tab-icon.svg";
import caretUp from "../../../assets/images/component-icons/caret-up.svg";
import caretDown from "../../../assets/images/component-icons/caret-down.svg";
import Section from "../../modules/section";

// proptypes
interface CommonSideNavLinkProps {
  href: string;
  text: string;
}

// Conditional Props:
// External links - If isExternal is set to true, then IconSvg & isSection props cannot be set
// Internal links - If isExternal is set to false, then IconSvg & isSection props are required
type ConditionalSideNavLinkProps =
  | {
      isExternal: false;
      IconSvg: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
      isSection: boolean;
    }
  | {
      isExternal: true;
      IconSvg?: never;
      isSection?: never;
    };

// Combined type of common and conditional props
export type SideNavLinkProps = CommonSideNavLinkProps &
  ConditionalSideNavLinkProps;

interface SideNavProps {
  sectionLinks: SideNavLinkProps[];
  pages: Page[];
  mobileItems: MobileItemProps;
}

// helpers
const bySection = (section: string) => (d) =>
  section && d.id.indexOf(`${section}/`) === 0;

const getPagesInSection = (pages: Page[], section: SideNavLinkProps) =>
  pages.filter(bySection(section.href));

const getCategory = (d) => (d.group ? d.group.category : d.category);

const findSelectedSection = (
  sectionLinks: SideNavLinkProps[],
  pathname: string
) =>
  sectionLinks.find(
    (section) =>
      pathname === section.href || pathname.startsWith(`${section.href}/`)
  );

const keys = (o: Record<string, {}>): string[] => {
  const sorted = Object.keys(o).sort();
  return sorted.indexOf("null") >= 0 || sorted.indexOf(null) >= 0
    ? [null, ...sorted.filter((s) => s !== "null" && s !== null)]
    : sorted;
};

const byDocGroup = (g: string) => (d: Page) =>
  (!d.group && typeof g === "undefined") || (d.group && d.group.id === g);

const groupsFrom = (categoryDocs: Page[]): PageGroup[] =>
  uniqBy(
    categoryDocs.map((c) => c.group),
    (g) => g?.id
  );

const byGuideStep = (a, b) =>
  a.guide && b.guide ? a.guide.step - b.guide.step : 0;

const sortCategories = (sectionHref: string, categories: string[]) => {
  const categoryOrder = Section.categories(sectionHref);
  return sortBy(categories, (c) => categoryOrder.indexOf(c));
};

const sortByWeight = (items) => sortBy(items, (i) => i?.weight);

// components
const Container = styled.div`
  position: relative;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  overflow: hidden;
`;

const Slide = styled.div`
  flex-grow: 1;
  display: flex;
  overflow: hidden;
`;

const SlidePane = styled.div`
  align-self: stretch;
  flex: 1 1 auto;
  overflow-x: hidden;
  overflow-y: auto;

  &.left {
    animation: ${slideInFromLeft} 400ms ease;
  }

  &.right {
    animation: ${slideInFromRight} 400ms ease;
  }
`;

const SectionWrap = styled.div`
  @media (${breakDown("md")}) {
    padding-top: 17px;
  }
`;

const ExternalSectionWrap = styled.div`
  border-top: 1px solid ${LAYOUT_BORDER};
`;

const StickySectionWrap = styled.div`
  position: sticky;
  top: 0;
  background-color: ${WHITE_PRIMARY};

  @media (${breakDown("md")}) {
    padding-top: 17px;
  }
`;

const SectionIconWrap = styled.div`
  width: 17px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 20px;

  svg {
    * {
      fill: ${PURPLE_PRIMARY};
    }
  }
`;

const SectionCaretWrap = styled.div`
  display: flex;
  justify-content: flex-end;
  flex-grow: 1;

  @media (${breakDown("md")}) {
    display: none;
  }
`;

type SectionLinkProps = {
  linkProps: SideNavLinkProps;
  isActive: boolean;
};

function SectionLink({ linkProps, isActive }: SectionLinkProps) {
  const { href, IconSvg, isSection, text, isExternal } = linkProps;

  return (
    <Link href={href} passHref>
      <a
        target={isExternal ? "_blank" : undefined}
        className={classnames({ active: isActive })}
        css={css`
          display: flex;
          height: 53px;
          padding: 0 15px;
          margin-left: 3px;
          align-items: center;
          justify-content: flex-start;
          font-family: ${ROBOTO};
          font-weight: 400;
          font-size: 14px;
          line-height: 16px;
          color: ${PURPLE_PRIMARY};
          text-decoration: none;

          &:hover,
          &:focus {
            margin-left: unset;
            background: linear-gradient(${GREY_HOVER}, ${GREY_HOVER})
                padding-box,
              linear-gradient(${VERTICAL_GRADIENT}) border-box;
            border-left: 3px solid transparent;
            outline: none;
          }

          &.active {
            margin-left: unset;
            background: linear-gradient(${GREY_HOVER}, ${GREY_HOVER})
                padding-box,
              linear-gradient(${VERTICAL_GRADIENT}) border-box;
            border-left: 3px solid transparent;
          }
        `}
      >
        <SectionIconWrap>
          {!isExternal ? (
            <IconSvg className="section-icon" width={17} height={17} />
          ) : (
            <NewTabIcon className="section-icon" width={15} height={15} />
          )}
        </SectionIconWrap>

        {text}

        {isSection && (
          <SectionCaretWrap>
            <RightIcon width={5} height={10} />
          </SectionCaretWrap>
        )}
      </a>
    </Link>
  );
}

const CategoriesWrap = styled.ul`
  margin: 10px 0 0 34px;
  padding: 0;
  border-left: 1px solid ${GREY_2};
`;

const Category = styled.li`
  list-style-type: none;
`;

const CategoryHeading = styled.div`
  text-transform: uppercase;
  font-family: ${POPPINS};
  font-weight: 600;
  color: #354153;
  font-size: 11px;
  margin: 20px 10px 10px;
`;

const groupToggleDocLinkStyles = css`
  margin: 6px 0;
  font-size: 14px;
  line-height: 140%;
  font-family: ${ROBOTO};
  color: ${PURPLE_PRIMARY};
  padding: 4px 20px 4px 10px;
  border-left: 1px solid transparent;
  margin-left: -1px;

  &:hover,
  &:focus {
    outline: none;
    margin-left: -1px;
    background-color: ${GREY_HOVER};
  }
`;

const GroupToggle = styled.div`
  ${groupToggleDocLinkStyles}
  display: flex;
  justify-content: space-between;
  align-items: center;
  cursor: pointer;
`;

const GroupCaret = styled.img`
  max-width: 7px;
`;

interface DocLinkProps {
  active: boolean;
  children: string;
  grouped: boolean;
  href: string;
}

function DocLink(props: DocLinkProps) {
  return (
    <Link href={props.href} passHref>
      <a
        css={css`
          ${groupToggleDocLinkStyles}
          display: block;
          text-decoration: none;

          &.active {
            background: linear-gradient(${GREY_HOVER}, ${GREY_HOVER})
                padding-box,
              linear-gradient(${VERTICAL_GRADIENT}) border-box;
            border-left: 1px solid transparent;
            color: ${ORANGE_PRIMARY};
          }

          &.grouped {
            padding-left: 30px;
          }
        `}
        className={classnames({ active: props.active, grouped: props.grouped })}
      >
        {props.children}
      </a>
    </Link>
  );
}

type DocGroupProps = {
  title: string;
  docs: Page[];
};

function DocGroup({ title, docs }: DocGroupProps) {
  const { pathname } = useRouter();
  const [isExpanded, setIsExpanded] = useState(
    !!docs.find((d) => d.id === pathname)
  );
  useEffect(() => {
    setIsExpanded(docs.find((d) => d.id === pathname) ? true : isExpanded);
  }, [pathname]);

  return (
    <>
      <GroupToggle
        tabIndex={0}
        onClick={() => setIsExpanded(!isExpanded)}
        onKeyPress={(e) =>
          e.key === "Enter" ? setIsExpanded(!isExpanded) : false
        } // eslint-disable-line react/jsx-curly-newline
      >
        {title}

        {isExpanded ? (
          <GroupCaret src={caretUp} alt="" />
        ) : (
          <GroupCaret src={caretDown} alt="" />
        )}
      </GroupToggle>

      {isExpanded &&
        docs.sort(byGuideStep).map((d) => (
          <DocLink key={d.id} grouped href={d.id} active={d.id === pathname}>
            {d.title}
          </DocLink>
        ))}
    </>
  );
}

const MobileWrap = styled.div`
  margin-top: 17px;
  padding-top: 17px;
  border-top: 1px solid ${LAYOUT_BORDER};

  @media (${breakUp("lg")}) {
    display: none;
  }
`;

const MobileButtonWrap = styled.div`
  padding: 17px 20px;
`;

function MobileItems({ button }: MobileItemProps) {
  return (
    <MobileButtonWrap>
      <Button
        variant="contained"
        href={button.link.href}
        target={button.link.external ? "_blank" : undefined}
        color="secondary"
        size="medium"
        fullWidth
      >
        {button.text}
      </Button>
    </MobileButtonWrap>
  );
}

function SideNav({ sectionLinks, pages, mobileItems }: SideNavProps) {
  const { pathname } = useRouter();
  const [activeSection, setActiveSection] = useState(
    findSelectedSection(sectionLinks, pathname)
  );
  useEffect(() => {
    setActiveSection(findSelectedSection(sectionLinks, pathname));
  }, [pathname]);

  const categories = activeSection
    ? groupby(getPagesInSection(pages, activeSection), getCategory)
    : {};

  return (
    <Container>
      <Slide>
        <SlidePane
          id="body-scroll-lock-side-nav"
          className={
            !(activeSection && activeSection.isSection) ? "left" : "right"
          }
        >
          {!(activeSection && activeSection.isSection) ? (
            <>
              <SectionWrap>
                {/* Filter sectionLinks based on the 'isExternal' prop so that only internal links
              are passed to the SectionLink component*/}
                {sectionLinks
                  .filter((l) => !l.isExternal)
                  .map((l) => (
                    <SectionLink
                      key={l.href}
                      isActive={l.href === pathname}
                      linkProps={l}
                    />
                  ))}
              </SectionWrap>
              <ExternalSectionWrap>
                {/* Filter sectionLinks based on the 'isExternal' prop so that only external links
             are passed to the SectionLink component */}
                {sectionLinks
                  .filter((el) => el.isExternal)
                  .map((el) => (
                    <SectionLink
                      key={el.href}
                      isActive={el.href === pathname}
                      linkProps={el}
                    />
                  ))}
              </ExternalSectionWrap>
            </>
          ) : (
            <>
              <StickySectionWrap>
                <SectionLink
                  linkProps={{
                    href: "/",
                    IconSvg: BackIcon,
                    isSection: false,
                    text: "Back",
                    isExternal: false,
                  }}
                  isActive={false}
                />

                <SectionLink
                  linkProps={{ ...activeSection, isSection: false }}
                  isActive
                />
              </StickySectionWrap>

              <CategoriesWrap>
                {sortCategories(activeSection.href, keys(categories)).map(
                  (c) => {
                    const categoryDocs = categories[c];
                    const groups = groupsFrom(categoryDocs);

                    return (
                      <Category key={c}>
                        {c !== "undefined" && (
                          <CategoryHeading>{c}</CategoryHeading>
                        )}

                        {sortByWeight(groups).map((g?: PageGroup) => {
                          const docs = categoryDocs.filter(byDocGroup(g?.id));
                          const ungroupedGroup = typeof g === "undefined";

                          return (
                            <div key={g?.id}>
                              {ungroupedGroup ? (
                                sortByWeight(docs).map((d: Page) => (
                                  <DocLink
                                    key={d.id}
                                    grouped={false}
                                    href={d.id}
                                    active={d.id === pathname}
                                  >
                                    {d.title}
                                  </DocLink>
                                ))
                              ) : (
                                <DocGroup title={g?.title} docs={docs} />
                              )}
                            </div>
                          );
                        })}
                      </Category>
                    );
                  }
                )}
              </CategoriesWrap>
            </>
          )}

          <MobileWrap>
            <MobileItems {...mobileItems} />
          </MobileWrap>
        </SlidePane>
      </Slide>
    </Container>
  );
}

export default SideNav;
