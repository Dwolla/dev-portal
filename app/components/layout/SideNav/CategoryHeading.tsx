import styled from "@emotion/styled";
import { POPPINS } from "../../typography";

export const CategoryHeading = styled.div`
  text-transform: uppercase;
  font-family: ${POPPINS};
  font-weight: 600;
  color: #354153;
  font-size: 11px;
  margin: 20px 10px 10px;
`;

export default CategoryHeading;
