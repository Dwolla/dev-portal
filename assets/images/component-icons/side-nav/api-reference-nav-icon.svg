<?xml version="1.0" encoding="UTF-8"?>
<svg width="13px" height="17px" viewBox="0 0 12 17" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 61.1 (101010) - https://sketch.com -->
    <title>B193B71A-BF48-4195-A7F9-D51C82930247</title>
    <desc>Created with sketchtool.</desc>
    <g id="Pages" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="SDKs-&amp;-Tools" transform="translate(-32.000000, -148.000000)" fill="#52627B" fill-rule="nonzero">
            <g id="Side-Nav-3" transform="translate(30.000000, 0.000000)">
                <g id="api-reference-icon-48x48" transform="translate(2.000000, 148.000000)">
                    <path d="M12.9954118,4.04267391 C12.9932263,4.0099339 12.9865401,3.97762095 12.9755294,3.94658696 C12.9713235,3.93476087 12.9682647,3.92367391 12.9629118,3.91221739 C12.944302,3.87200659 12.9184044,3.83533547 12.8864412,3.80393478 L9.06291176,0.108282609 C9.03044107,0.0780166337 8.99262959,0.0536103283 8.95126471,0.0362173913 C8.93879346,0.0311838212 8.92602624,0.0268647453 8.91302941,0.0232826087 C8.88188984,0.013063135 8.84949388,0.00685034149 8.81667647,0.00480434783 C8.80926698,0.00275318105 8.80173288,0.00114864507 8.79411765,0 L0.382352941,0 C0.171185243,0 0,0.165459984 0,0.369565217 L0,16.6304348 C0,16.83454 0.171185243,17 0.382352941,17 L12.6176471,17 C12.8288148,17 13,16.83454 13,16.6304348 L13,4.06521739 C13,4.05745652 12.9957941,4.05043478 12.9954118,4.04267391 Z M9.17647059,1.26169565 L11.6946471,3.69565217 L9.17647059,3.69565217 L9.17647059,1.26169565 Z M12.2352941,16.2608696 L0.764705882,16.2608696 L0.764705882,0.739130435 L8.41176471,0.739130435 L8.41176471,4.06521739 C8.41176471,4.26932262 8.58294995,4.43478261 8.79411765,4.43478261 L12.2352941,4.43478261 L12.2352941,16.2608696 Z" id="Shape"></path>
                    <path d="M7.78164706,5.93632609 C7.68663553,5.90172895 7.58128872,5.90507766 7.48883431,5.94563385 C7.3963799,5.98619005 7.32441037,6.06062321 7.28879412,6.15252174 L4.99467647,12.0655652 C4.9588822,12.1573991 4.96234678,12.2592226 5.00430631,12.3485849 C5.04626583,12.4379472 5.12327454,12.5075097 5.21835294,12.5419348 C5.26148439,12.5571627 5.30702275,12.5650404 5.35294118,12.5652176 C5.5123318,12.5653714 5.65510044,12.4699392 5.71120588,12.3257391 L8.00532353,6.41269565 C8.0411178,6.32086177 8.03765322,6.21903826 7.99569369,6.12967597 C7.95373417,6.04031368 7.87672546,5.97075116 7.78164706,5.93632609 Z" id="Path"></path>
                    <path d="M4.93235294,7.59567391 C4.88705638,7.50792119 4.80753175,7.44116824 4.71129121,7.41011477 C4.61505068,7.37906129 4.50998663,7.3862539 4.41923529,7.4301087 L1.36041176,8.90836957 C1.23063563,8.97088651 1.14861352,9.0989787 1.14861352,9.23913043 C1.14861352,9.37928217 1.23063563,9.50737436 1.36041176,9.5698913 L4.41923529,11.0481522 C4.54149368,11.1071709 4.68704069,11.0986638 4.80105017,11.0258354 C4.91505965,10.953007 4.98021085,10.8269216 4.97196194,10.6950745 C4.96371303,10.5632274 4.88331721,10.4456492 4.76105882,10.3866304 L2.38435294,9.23913043 L4.75914706,8.09347826 C4.85077208,8.05003333 4.92062524,7.97298657 4.95315669,7.8794892 C4.98568813,7.78599183 4.97819774,7.68380414 4.93235294,7.59567391 L4.93235294,7.59567391 Z" id="Path"></path>
                    <path d="M8.06955882,7.59567391 C8.02418656,7.68339009 8.01674508,7.78494029 8.04887306,7.87796208 C8.08100105,7.97098386 8.1500638,8.04784881 8.24085294,8.09163043 L10.6156471,9.23913043 L8.24085294,10.3866304 C8.05185786,10.4778655 7.97516687,10.6999128 8.06955883,10.882587 C8.16395079,11.0652611 8.39368138,11.1393872 8.58267647,11.0481522 L11.6415,9.5698913 C11.7712761,9.50737436 11.8532982,9.37928217 11.8532982,9.23913043 C11.8532982,9.0989787 11.7712761,8.97088651 11.6415,8.90836957 L8.58267647,7.4301087 C8.49192513,7.3862539 8.38686108,7.37906129 8.29062055,7.41011477 C8.19438002,7.44116824 8.11485538,7.50792119 8.06955882,7.59567391 L8.06955882,7.59567391 Z" id="Path"></path>
                </g>
            </g>
        </g>
    </g>
</svg>