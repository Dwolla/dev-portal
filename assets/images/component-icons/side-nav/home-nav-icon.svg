<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15 17">
  <path d="M14.6 5.122L8.1.203a1 1 0 00-1.2 0L.4 5.122a1 1 0 00-.4.8v9a2 2 0 002 2h11a2 2 0 002-2v-9a1 1 0 00-.4-.8zM6 15.922v-7h3v7H6zm8-1a1 1 0 01-1 1h-3v-7a1 1 0 00-1-1H6a1 1 0 00-1 1v7H2a1 1 0 01-1-1v-9L7.5 1 14 5.922v9z" fill="#52627B" fill-rule="nonzero"/>
</svg>