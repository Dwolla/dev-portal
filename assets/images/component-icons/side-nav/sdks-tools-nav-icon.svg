<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15 17">
  <path d="M14.5 3.884L8 .134a1 1 0 00-1 0L.5 3.884a1 1 0 00-.5.866v7.5a1 1 0 00.5.866l6.5 3.75a1 1 0 001 0l6.5-3.75a1 1 0 00.5-.866v-7.5a1 1 0 00-.5-.866zM7.5 1l6.029 3.479L7.5 7.924 1.47 4.479 7.5 1zM1 5.362L7 8.79v6.922L1 12.25V5.362zm7 10.35V8.79l6-3.429v6.889l-6 3.462z" fill="#52627B" fill-rule="nonzero"/>
</svg>