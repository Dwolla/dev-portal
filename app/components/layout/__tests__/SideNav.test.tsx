import renderer from "react-test-renderer";
import SideNav, { SideNavLinkProps } from "../SideNav";
import Pages from "../../../modules/pages";

import { ReactComponent as HomeIcon } from "../../../../assets/images/component-icons/side-nav/home-nav-icon.svg";
import { ReactComponent as DwollaBalanceIcon } from "../../../../assets/images/component-icons/side-nav/dwolla-balance-nav-icon.svg";
import { ReactComponent as SdksToolsIcon } from "../../../../assets/images/component-icons/side-nav/sdks-tools-nav-icon.svg";
import { ReactComponent as ApiReferenceIcon } from "../../../../assets/images/component-icons/side-nav/api-reference-nav-icon.svg";

const mockPath = "/";
jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: mockPath,
      pathname: mockPath,
      query: "",
      asPath: mockPath,
    };
  },
}));
jest.mock("../../../modules/images.import");
jest.mock("../../../modules/pages.import");

const SIDE_NAV_LINKS: SideNavLinkProps[] = [
  {
    href: "/docs",
    IconSvg: HomeIcon,
    isSection: false,
    text: "Home",
    isExternal: false,
  },
  {
    href: "/docs/balance",
    IconSvg: DwollaBalanceIcon,
    isSection: false,
    text: "Balance",
    isExternal: false,
    stickyReferenceLinks: [
      {
        href: "/api-reference",
        IconSvg: ApiReferenceIcon,
        isSection: false,
        text: "API Reference",
        isExternal: false,
      },
    ],
  },
  {
    href: "/api-reference",
    IconSvg: ApiReferenceIcon,
    isSection: true,
    text: "API Reference",
    isExternal: false,
  },
  {
    href: "/sdks-tools",
    IconSvg: SdksToolsIcon,
    isSection: false,
    text: "SDKs & Tools",
    isExternal: false,
  },
  {
    href: "https://discuss.dwolla.com/",
    text: "Community",
    isExternal: true,
  },
];

const TOP_BAR_PROPS = {
  button: {
    text: "Get API Keys",
    link: {
      href: "https://accounts-sandbox.dwolla.com/login",
      external: true,
    },
  },
};

const PRODUCT_OPTIONS = [
  { value: "connect", label: "Dwolla Connect" }, // Array[0] is selected by default
  { value: "balance", label: "Dwolla Balance" },
];

test("SideNav", () => {
  const tree = renderer
    .create(
      <SideNav
        pages={Pages.all()}
        sectionLinks={SIDE_NAV_LINKS}
        productSelectorOptions={PRODUCT_OPTIONS}
        mobileItems={TOP_BAR_PROPS}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
