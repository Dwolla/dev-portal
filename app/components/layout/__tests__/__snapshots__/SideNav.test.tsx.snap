// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav 1`] = `
<div
  className="css-yeohq5-Container e1qiqw1s14"
>
  <div
    className="css-kibqvq-Slide e1qiqw1s13"
  >
    <div
      className="left css-1jc4k7f-SlidePane e1qiqw1s12"
      id="body-scroll-lock-side-nav"
    >
      <div
        className="css-1lugamu-SectionWrap e1qiqw1s11"
      >
        <a
          className="active css-yka8go-SectionLink"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="css-1uyhva7-SectionIconWrap e1qiqw1s8"
          >
            <svg
              className="section-icon"
              height={17}
              width={17}
            />
          </div>
          Home
        </a>
        <a
          className=" css-yka8go-SectionLink"
          href="/api-reference"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="css-1uyhva7-SectionIconWrap e1qiqw1s8"
          >
            <svg
              className="section-icon"
              height={17}
              width={17}
            />
          </div>
          API Reference
          <div
            className="css-y8ea5i-SectionCaretWrap e1qiqw1s7"
          >
            <svg
              height={10}
              width={5}
            />
          </div>
        </a>
        <a
          className=" css-yka8go-SectionLink"
          href="/guides"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="css-1uyhva7-SectionIconWrap e1qiqw1s8"
          >
            <svg
              className="section-icon"
              height={17}
              width={17}
            />
          </div>
          Guides
          <div
            className="css-y8ea5i-SectionCaretWrap e1qiqw1s7"
          >
            <svg
              height={10}
              width={5}
            />
          </div>
        </a>
        <a
          className=" css-yka8go-SectionLink"
          href="/concepts"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="css-1uyhva7-SectionIconWrap e1qiqw1s8"
          >
            <svg
              className="section-icon"
              height={17}
              width={17}
            />
          </div>
          Concepts
          <div
            className="css-y8ea5i-SectionCaretWrap e1qiqw1s7"
          >
            <svg
              height={10}
              width={5}
            />
          </div>
        </a>
        <a
          className=" css-yka8go-SectionLink"
          href="/sdks-tools"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <div
            className="css-1uyhva7-SectionIconWrap e1qiqw1s8"
          >
            <svg
              className="section-icon"
              height={17}
              width={17}
            />
          </div>
          SDKs & Tools
        </a>
      </div>
      <div
        className="css-10a5zt7-ExternalSectionWrap e1qiqw1s10"
      >
        <a
          className=" css-yka8go-SectionLink"
          href="https://discuss.dwolla.com/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          target="_blank"
        >
          <div
            className="css-1uyhva7-SectionIconWrap e1qiqw1s8"
          >
            <svg
              className="section-icon"
              height={15}
              width={15}
            />
          </div>
          Community
        </a>
      </div>
      <div
        className="css-nasyo1-MobileWrap e1qiqw1s1"
      >
        <div
          className="css-1rc904k-MobileButtonWrap e1qiqw1s0"
        >
          <a
            className="primary block  css-ux75yb-StyledLink-buttonStyles-buttonStyles ek3ainu0"
            href="https://accounts-sandbox.dwolla.com/login"
            target="_blank"
          >
            Get API Keys
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
